pkg/control/pubcontrol/pub_control_utils.go

```diff
@@ -102,7 +102,7 @@ func PodUnavailableBudgetValidatePod(pod *corev1.Pod, operation policyv1alpha1.P
 		if refresh {
 			pubClone, err = kubeClient.GetGenericClient().KruiseClient.PolicyV1alpha1().
 				PodUnavailableBudgets(pub.Namespace).Get(context.TODO(), pub.Name, metav1.GetOptions{})
-			if err != nil {
+			if err != nil && !errors.IsNotFound(err) {
```

Comment:
if errors.IsNotFound(err), code will goes to the else clause, can the code just return so as to avoid unnecessary and potentially problematic code?

Reply:
Do you mean to return nil directly?

Reply:
We also should check error returned in pub status updating,  just ingore error if NotFound error retured.  https://github.com/openkruise/kruise/blob/142458151be55a538c78c80c6c791cfe9c1fdc10/pkg/control/pubcontrol/pub_control_utils.go#L165

Reply:
@Spground  
Adding this feels a bit redundant since PUB exits without being discovered and does not continue running kclient.Status().Update

If you need to add it, is the following suitable?
```go
                err = kclient.Status().Update(context.TODO(), pubClone)
                costOfUpdate += time.Since(start)
                if err == nil {
                        if err = util.GlobalCache.Add(pubClone); err != nil {
                                klog.ErrorS(err, "Failed to add cache for podUnavailableBudget", "pub", klog.KObj(pub))
                        }
                        return nil
                } else {
                        if errors.IsNotFound(err) {
                                return nil
                        }
                }
```

Reply:
> @Spground Adding this feels a bit redundant since PUB exits without being discovered and does not continue running kclient.Status().Update
> 
> If you need to add it, is the following suitable?
> 
> ```go
>                 err = kclient.Status().Update(context.TODO(), pubClone)
>                 costOfUpdate += time.Since(start)
>                 if err == nil {
>                         if err = util.GlobalCache.Add(pubClone); err != nil {
>                                 klog.ErrorS(err, "Failed to add cache for podUnavailableBudget", "pub", klog.KObj(pub))
>                         }
>                         return nil
>                 } else {
>                         if errors.IsNotFound(err) {
>                                 return nil
>                         }
>                 }
> ```

It's not exactly.  PUB can be deleted anytime, e.g,  before you UPDATE Status and after GET pub.

Reply:
@Spground  ok,IsNotFound has been added to both locations. Do I still need get informerCached to make IsNotFound judgment?